Mathlib/Lean/Expr/Lit.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2025 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Qq
+
+/-!
+# Extract literals from expressions
+
+
+## TODO
+
+Redefine
+```lean4
+def natLit! (e : Expr) : Nat :=
+  match e.natLit? with
+  | some n => n
+  | none => panic! "not a raw natural literal"
+
+def intLit! (e : Expr) : Int :=
+  match e.intLit? with
+  | some n => n
+  | none => panic! "not a raw integer literal"
+```
+-/
+
+open Lean
+
+namespace Lean.Expr
+
+/-- Returns the natural number from a natural literal expression, or `none` if the expression isn't
+a natural literal.
+
+See `Qq.natLitQq?` for a Qq version of this that also handles `OfNat.ofNat`. -/
+def natLit? : Expr → Option Nat
```

Comment:
This is `rawNatLit?`

Reply:
I am aware, but this name matches `Lean.Expr.natLit!` and should therefore replace `rawNatLit?`, unless you think there's a better reason to include "raw" in the name?

Reply:
The reason would `Lean.mkNatLit` vs `Lean.mkRawNatLit`, where the former has `OfNat.ofNat` and the latter does not.